src/python/zquantum/vqe/singlet_uccsd_test.py

```diff
@@ -0,0 +1,124 @@
+                transformation=self.transformation,
+            )
+
+    def test_set_number_of_alpha_electrons(self):
+        # Given
+        new_number_of_alpha_electrons = 2
+
+        # When
+        self.ansatz.number_of_alpha_electrons = new_number_of_alpha_electrons
+
+        # Then
+        self.assertEqual(
+            self.ansatz._number_of_alpha_electrons, new_number_of_alpha_electrons
+        )
+
+    def test_set_number_of_alpha_electrons_asserts_number_of_spatial_orbitals(self):
+        # Given
+        new_number_of_alpha_electrons = 3
+
+        # When/Then
+        with self.assertRaises(ValueError):
+            self.ansatz.number_of_alpha_electrons = new_number_of_alpha_electrons
+
+    def test_get_number_of_beta_electrons(self):
+        # Given
+        new_number_of_alpha_electrons = 2
+
+        # When
+        self.ansatz.number_of_alpha_electrons = new_number_of_alpha_electrons
+
+        # Then
+        self.assertEqual(
+            self.ansatz._number_of_beta_electrons, new_number_of_alpha_electrons
+        )
+
+    def test_get_number_of_qubits(self):
+        # Given
+        new_number_of_spatial_orbitals = 4
+        target_number_of_qubits = 8
+
+        # When
+        self.ansatz.number_of_spatial_orbitals = new_number_of_spatial_orbitals
+
+        # Then
+        self.assertEqual(self.ansatz.number_of_qubits, target_number_of_qubits)
+
+    def test_set_number_of_spatial_orbitals(self):
+        # Given
+        new_number_of_spatial_orbitals = 4
+
+        # When
+        self.ansatz.number_of_spatial_orbitals = new_number_of_spatial_orbitals
+
+        # Then
+        self.assertEqual(
+            self.ansatz._number_of_spatial_orbitals, new_number_of_spatial_orbitals
+        )
+
+    def test_set_number_of_spatial_orbitals_asserts_when_smaller_than_2(self):
+        # Given
+        new_number_of_spatial_orbitals = 1
+
+        # When/Then
+        with self.assertRaises(ValueError):
+            self.ansatz.number_of_spatial_orbitals = new_number_of_spatial_orbitals
+
+    def test_set_number_of_spatial_orbitals_asserts_when_smaller_than_number_of_alpha_electrons(
+        self,
+    ):
+        # Given
+        self.ansatz.number_of_spatial_orbitals = 4
+        self.ansatz.number_of_alpha_electrons = 3
+        new_number_of_spatial_orbitals = 3
+
+        # When/Then
+        with self.assertRaises(ValueError):
+            self.ansatz.number_of_spatial_orbitals = new_number_of_spatial_orbitals
+
+    def test_set_transformation(self):
+        # Given
+        new_transformation = "transformation"
+
+        # When
+        self.ansatz.transformation = new_transformation
+
+        # Then
+        self.assertEqual(self.ansatz._transformation, new_transformation)
+
+    def test_does_not_support_parametrized_circuit(self):
+        # When/Then
+        self.assertFalse(self.ansatz.supports_parametrized_circuits)
+        with self.assertRaises(Exception):
```

Comment:
I don't think so, since it's not matter of having it "not implemented", rather a more fundamental issue with not being able to use parametrizable circuit here (due to issues with openfermion in this case).
So I would rather leave it with a more general Exception.

Reply:
See: [6f9060](https://github.com/zapatacomputing/z-quantum-vqe/pull/5/commits/6f9060106f5706db219da2bd6062cfaa2fd485e3).